ecommerce/pricing/lib/pricing/services.rb

```diff
@@ -45,7 +45,31 @@ def initialize(event_store)
 
     def call(cmd)
       @repository.with_aggregate(Offer, cmd.aggregate_id) do |order|
-        order.change_discount(Discounts::PercentageDiscount.new(cmd.amount))
+        order.change_discount(Discounts::GENERAL_DISCOUNT, Discounts::PercentageDiscount.new(cmd.amount))
+      end
+    end
+  end
+
+  class SetTimePromotionDiscountHandler
+    def initialize(event_store)
+      @repository = Infra::AggregateRootRepository.new(event_store)
+    end
+
+    def call(cmd)
+      @repository.with_aggregate(Offer, cmd.aggregate_id) do |order|
+        order.apply_discount(Discounts::TIME_PROMOTION_DISCOUNT, Discounts::PercentageDiscount.new(cmd.amount))
+      end
+    end
+  end
+
+  class ResetTimePromotionDiscountHandler
+    def initialize(event_store)
+      @repository = Infra::AggregateRootRepository.new(event_store)
+    end
+
+    def call(cmd)
+      @repository.with_aggregate(Offer, cmd.aggregate_id) do |order|
+        order.reset_discount(Discounts::TIME_PROMOTION_DISCOUNT)
```

Comment:
Is it resetting the time promotion discount or is it just removing (time promotion) discount from the order?

Reply:
It is removing. I used the `reset_discount` method that already was there but you are right that maybe renaming it to `remove_discount` would better represent what it does.